### PR TITLE
fix annotations test flake

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://github.com/openshift/origin/issues/11442

@danwinship @openshift/cli-review

### CODE REVIEW

File: test/cmd/annotations.sh [hunk 1]
```diff
@@ -13,16 +13,16 @@ trap os::test::junit::reconcile_output EXIT
 os::test::junit::declare_suite_start "cmd/annotate"
 # This test validates empty values in key-value pairs set by the annotate command
 os::cmd::expect_success_and_text 'oc process -f examples/zookeeper/template.json | oc apply -f -' 'pod "zookeeper-1" created'
-os::cmd::expect_success_and_text 'oc annotate pod zookeeper-1 node-selector=""' 'pod "zookeeper-1" annotated'
-os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.annotations}}"' 'node-selector:'
+os::cmd::expect_success_and_text 'oc annotate pod zookeeper-1 nodeselector=""' 'pod "zookeeper-1" annotated'
```
Comment (stevekuznetsov) [inline test/cmd/annotations.sh]: This change seems unrelated?

File: test/cmd/annotations.sh [hunk 2]
```diff
@@ -13,16 +13,16 @@ trap os::test::junit::reconcile_output EXIT
...
+os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.annotations.nodeselector}}"' ''
 
 echo "annotate: ok"
 os::test::junit::declare_suite_end
 
 os::test::junit::declare_suite_start "cmd/label"
 # This test validates empty values in key-value pairs set by the label command
 os::cmd::expect_success_and_text 'oc label pod zookeeper-1 label2=""' 'pod "zookeeper-1" labeled'
-os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.labels}}"' 'label2\: '
+os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.labels.label2}}"' ''
```
Comment (stevekuznetsov) [inline test/cmd/annotations.sh]: It is not obvious to me that this will test what you want it to ... I think we need to be explicit in the expected text:
```
os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.labels.label2}}"' '^$'
```

File: test/cmd/annotations.sh (hunk 1 above)
Comment (juanvallejo) [inline test/cmd/annotations.sh]: It is, but it was not testing an empty value annotation well enough before. The line below it actually checks for the annotation's value now, as opposed to just the annotation name and a colon

File: test/cmd/annotations.sh (hunk 2 above)
Comment (liggitt) [inline test/cmd/annotations.sh]: if you want to ensure the empty value, you need to pin to start and end: '^$'
if the label is completely missing, you'll get '<no value>' from the template output and '' will match that since it does substring regex match

Comment (juanvallejo) [inline test/cmd/annotations.sh]: Sounds good, will update

File: test/cmd/annotations.sh [hunk 3]
```diff
@@ -13,16 +13,16 @@ trap os::test::junit::reconcile_output EXIT
 os::test::junit::declare_suite_start "cmd/annotate"
 # This test validates empty values in key-value pairs set by the annotate command
 os::cmd::expect_success_and_text 'oc process -f examples/zookeeper/template.json | oc apply -f -' 'pod "zookeeper-1" created'
-os::cmd::expect_success_and_text 'oc annotate pod zookeeper-1 node-selector=""' 'pod "zookeeper-1" annotated'
-os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.annotations}}"' 'node-selector:'
+os::cmd::expect_success_and_text 'oc annotate pod zookeeper-1 nodeselector=""' 'pod "zookeeper-1" annotated'
+os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.annotations.nodeselector}}"' ''
```
Comment (liggitt) [inline test/cmd/annotations.sh]: same comment here, if you want to test empty value, pin '^$'

File: test/cmd/annotations.sh (hunk 1 above)
Comment (stevekuznetsov) [inline test/cmd/annotations.sh]: Right, but you can do that without renaming the annotation

**Comment by juanvallejo** [comment]
@stevekuznetsov @liggitt Thanks for the feedback, expecting `^$` was not working since the command output did not have a newline marker at the end

File: test/cmd/annotations.sh (hunk 1 above)
Comment (juanvallejo) [inline test/cmd/annotations.sh]: hm, I was getting `error: error parsing template {{.metadata.annotations.node-selector}}, template: output:1: bad character U+002D '-'` when I gave it the template `{{.metadata.annotations.node-selector`}}

**Comment by juanvallejo** [comment]
@stevekuznetsov or @liggitt do you have any more feedback? Using `expect_success_and_not_text 'oc get pod ... --template="{{.metadata.labels.label2}}' '.'` seems to be working well

File: test/cmd/annotations.sh (hunk 1 above)
Comment (stevekuznetsov) [inline test/cmd/annotations.sh]: I can't remember my `jsonpath` well -- is there a `object['field-name']` accessor?

Comment (juanvallejo) [inline test/cmd/annotations.sh]: hm, apparently I can use '{{index .metadata.annotations "my-index"}}' whenever a key includes "bad characters"

Comment (stevekuznetsov) [inline test/cmd/annotations.sh]: Let's do that

Comment (juanvallejo) [inline test/cmd/annotations.sh]: Sounds good https://github.com/openshift/origin/pull/11443/commits/a2057c2763361e08a312db8991d7962406583e1c

**Comment by stevekuznetsov** [comment]
Squash, LGTM

**Comment by stevekuznetsov** [comment]
[merge]
